apps/3000-home/pages/_document.js

```diff
@@ -0,0 +1,30 @@
+import Document, { Html, Head, Main, NextScript } from 'next/document'
+import {revalidate} from "@module-federation/nextjs-mf/utils";
+
+class MyDocument extends Document {
+  static async getInitialProps(ctx) {
+    const initialProps = await Document.getInitialProps(ctx);
+
+    ctx?.res?.on('finish', () => {
```

Comment:
We could. But need to find where the best place would be. I generally don't like implementing it automatically because users may wish to use some other revalidation tactic other than res finish. 

If we can test revalidation in prod builds and if we can await before get initial props then we can add revalidation strategies for different lifecycles